player/client.c

```diff
@@ -1278,6 +1278,12 @@ static void getproperty_fn(void *arg)
 
 int mpv_get_property(mpv_handle *ctx, const char *name, mpv_format format,
                      void *data)
+{
+    return mp_get_property(ctx, name, format, data, true);
+}
+
+int mp_get_property(mpv_handle *ctx, const char *name, mpv_format format,
+                    void *data, bool lock)
```

Comment:
i pushed a change that uses the appropriate option structs (`mp_vo_opts` and `macos_opts`) for property retrieval on init. it adds a bit more complexity to cocoa-cb because of the two types of property retrieval. is this way more agreeable? otherwise i don't know how else to do this.

Reply:
What exactly is the problem? Why did this change now?

Reply:
before window creation etc was done async on the main thread, so it was actually executed after the callback events (reconfigure/preinit). it works but imo isn't very nice. the actual problem is that i want to support the `--cocoa-force-dedicated-gpu`. i need to retrieve that property when the context is created, and that happens on the core/main thread.

it will probably also useful for `--wid`  embedding.